### DISCUSSION TITLE
How can other modules access the progress bar if they are lower than the progress bar?

### DISCUSSION THREAD

**Discussion by KingKDot** [Textualize/textual#4019] [Q&A]
For example if I have my file structure like this

```
C:.
|   main.py
|   
\---other
        to_do.py
```

But I only want to increase the progress bar when certain things are completed within to_do.py, how can I achieve this? the UI and all it's components are inside main.py and I'm unsure if I'm able to have to_do.py access any of the UI modules to update them (specifically the progress bar). If this sounds confusing it's probably because I'm also confused. If I need to explain better let me know. Thank you for the help.

**Comment by davep** [comment]
All of the widgets in your application will, if they are displayed, be in the DOM. From there you can perform queries and interact with them. It's *very* common in non-trivial applications to have widgets and screens and the app all defined in very different modules; you can see this sort of thing [in frogmouth, for example](https://github.com/Textualize/frogmouth/tree/main/frogmouth).

A good read through [the guide to Textual](https://textual.textualize.io/guide/) should give you all the grounding you need to get this going; [DOM queries](https://textual.textualize.io/guide/queries/), [events and messages](https://textual.textualize.io/guide/events/) and [reactivity](https://textual.textualize.io/guide/reactivity/) will play a large part in this. As you work through the guide, you'll eventually get to the guide to developing widgets [and how to coordinate them](https://textual.textualize.io/guide/widgets/#coordinating-widgets).
